washee_app/lib/features/unlock/presentation/widgets/start_wash.dart

```diff
@@ -108,22 +111,46 @@ class _StartWashState extends State<StartWash> {
       _isUnlockingMachine = true;
     });
     try {
-      fetchedMachine = await sl<UnlockUseCase>().call(UnlockParams(
-          machine: widget.currentMachine,
-          duration: Duration(hours: 2, minutes: 30)));
-      if (fetchedMachine == null) {
-        setState(() {
-          _isUnlockingMachine = false;
-        });
-        ErrorHandler.errorHandlerView(
-            context: context,
-            prompt: HTTPErrorPrompt(
-                message:
-                    "Det ser ud til, at du ikke har forbindelse til WasheeBox"));
+      var result = await sl<ConnectBoxWifiUsecase>().call(NoParams());
+
+      if (result) {
+        fetchedMachine = await sl<UnlockUseCase>().call(UnlockParams(
+            machine: widget.currentMachine,
+            duration: Duration(hours: 2, minutes: 30)));
+        await sl<DisconnectBoxWifiUsecase>().call(NoParams());
```

Comment:
Som sagt bryder jeg mig ikke heeeeelt om det her. Men vi må tage et ordentligt "design møde" og få klarlagt 100% hvad strukturen er for hvordan sådan noget her bør håndteres. Men det er en fed løsning som det står nu!